src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsExecution.java

```diff
@@ -151,25 +144,37 @@ private static ExternalWorkspaceProperty findNodeProperty(String diskPoolRefId,
             }
         }
 
-        String nodeName = node.getDisplayName();
         if (exwsNodeProperty == null) {
-            String message = format("There is no External Workspace config defined in Node '%s' config", nodeName);
-            throw new AbortException(message);
-        }
-        String nodeDiskPoolRefId = exwsNodeProperty.getDiskPoolRefId();
-        if (nodeDiskPoolRefId == null) {
-            String message = format("The Disk Pool Ref ID was not provided in Node '%s' config", nodeName);
-            throw new AbortException(message);
-        }
-        if (!nodeDiskPoolRefId.equals(diskPoolRefId)) {
-            String message = format("In Node '%s' config, the defined Disk Pool Ref ID '%s' does not match the one allocated by the exwsAllocate step '%s'", nodeName, nodeDiskPoolRefId, diskPoolRefId);
+            String message = format("There is no External Workspace config defined in Node '%s' config", node.getDisplayName());
             throw new AbortException(message);
         }
 
         return exwsNodeProperty;
     }
 
     /**
+     * Selects the {@link DiskPoolNode} that has the {@link DiskPoolNode#diskPoolRefId} equal to the given
+     * {@code diskPoolRefId} param.
+     *
+     * @param diskPoolRefId the disk pool reference id to be searching for
+     * @param diskPoolNodes the list of disk pools
+     * @return the Disk Pool that has the matching reference id, {@code null} otherwise
+     */
+    @CheckForNull
+    private static DiskPoolNode findDiskPoolNode(@Nonnull String diskPoolRefId,
+                                                 @Nonnull List<DiskPoolNode> diskPoolNodes) {
+        DiskPoolNode selected = null;
+        for (DiskPoolNode diskPoolNode : diskPoolNodes) {
```

Comment:
I don't expect users to have a lot of disk pools, so I don't see the need to complicate things with disk pool indexing.